packages/flutter_tools/lib/src/commands/migrate_resolve_conflicts.dart

```diff
@@ -0,0 +1,322 @@
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'project-directory',
+      help: 'The root directory of the flutter project.',
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'context-lines',
+      defaultsTo: '5',
+      help: 'The number of lines of context to show around the each conflict. Defaults to 5.',
+    );
+    argParser.addFlag(
+      'confirm-commit',
+      defaultsTo: true,
+      help: 'Indicates if proposed changes require user verification before writing to disk.',
+    );
+  }
+
+  final Logger logger;
+
+  final FileSystem fileSystem;
+
+  final Terminal terminal;
+
+  @override
+  final String name = 'resolve-conflicts';
+
+  @override
+  final String description = 'Prints the current status of the in progress migration.';
+
+  @override
+  String get category => FlutterCommandCategory.project;
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{};
+
+  static const String _conflictStartMarker = '<<<<<<<';
+  static const String _conflictDividerMarker = '=======';
+  static const String _conflictEndMarker = '>>>>>>>';
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    final String? projectDirectory = stringArg('project-directory');
+    final FlutterProjectFactory flutterProjectFactory = FlutterProjectFactory(logger: logger, fileSystem: fileSystem);
+    final FlutterProject project = projectDirectory == null
+      ? FlutterProject.current()
+      : flutterProjectFactory.fromDirectory(fileSystem.directory(projectDirectory));
+
+    Directory stagingDirectory = project.directory.childDirectory(kDefaultMigrateStagingDirectoryName);
+    final String? customStagingDirectoryPath = stringArg('staging-directory');
+    if (customStagingDirectoryPath != null) {
+      if (fileSystem.path.isAbsolute(customStagingDirectoryPath)) {
+        stagingDirectory = fileSystem.directory(customStagingDirectoryPath);
+      } else {
+        stagingDirectory = project.directory.childDirectory(customStagingDirectoryPath);
+      }
+    }
+    if (!stagingDirectory.existsSync()) {
+      logger.printStatus('No migration in progress. Start a new migration with:');
+      printCommandText('flutter migrate start', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    final File manifestFile = MigrateManifest.getManifestFileFromDirectory(stagingDirectory);
+    final MigrateManifest manifest = MigrateManifest.fromFile(manifestFile);
+
+    checkAndPrintMigrateStatus(manifest, stagingDirectory, logger: logger);
+
+    final List<String> conflictFiles = manifest.remainingConflictFiles(stagingDirectory);
+
+    terminal.usesTerminalUi = true;
+
+    for (int i = 0; i < conflictFiles.length; i++) {
+      final String localPath = conflictFiles[i];
+      final File file = stagingDirectory.childFile(localPath);
+      final List<String> lines = file.readAsStringSync().split('\n');
```

Comment:
Here (and throughout this PR) you assume that newlines are unix-style (`\n`) rather than Windows style (`'\r\n'`).

I don't know if we need to handle the difference, or if this would all just work if a conflicted file has windows style newlines. Have you considered this?

Reply:
This should handle it fine. We remove a `\n` in line 108, leaving a `\r` at the end of the line. Then we add it back in in `verifyAndCommit` The resulting line ending should still be `\r\n` for windows, and `\n` for unix.